lib/recurly/pricing/subscription/calculations.js

```diff
@@ -172,9 +173,15 @@ export default class Calculations {
       this.price.base.addons[addon.code] = price;
       this.price.addons[addon.code] = price; // DEPRECATED
 
-      const selected = find(this.items.addons, { code: addon.code });
-      if (selected) {
-        price = price * selected.quantity;
+      const selectedQuantity = find(this.items.addons, { code: addon.code })?.quantity;
+      if (selectedQuantity) {
+        if (isTieredAddOn(addon)) {
+          const currencyCode = this.pricing.currencyCode;
+          price = getTieredPricingTotal(addon, selectedQuantity, currencyCode).unitAmount;
```

Comment:
Since we only ever use the return value property of `unitAmount`, let's just have this function return that.